src/antares/model/study.py

```diff
@@ -265,11 +265,9 @@ def create_binding_constraint(
         equal_term_matrix: Optional[pd.DataFrame] = None,
         greater_term_matrix: Optional[pd.DataFrame] = None,
     ) -> BindingConstraint:
-        constraint = self._binding_constraints_service.create_binding_constraint(
+        return self._binding_constraints_service.create_binding_constraint(
             name, properties, terms, less_term_matrix, equal_term_matrix, greater_term_matrix
         )
-        self._binding_constraints[constraint.id] = constraint
```

Comment:
Why did you remove this line ?

Reply:
I've moved the tracking of binding constraints from the 'Study' object to the 'binding constraints service' to correctly handle the .ini file.

Reply:
Ok